python/housinginsights/sources/census.py

```diff
@@ -1,61 +1,123 @@
-
-
 class CensusApiConn(BaseApiConn):
     """
-    Census API connector, confined to ACS5 2015 for now.
-
+    
     """
-    BASEURL = 'http://api.census.gov/data'
-
-    def __init__(self, arg):
-        super(CensusApiConn, self).__init__(CensusApiConn.BASEURL)
-        self.arg = arg
-
-    def getacs5(self):
-        params = {'key': census_key, 'get': 'B01003_001E,B25057_001E,B25058_001E,B25059_001E', 'for': 'tract:*', 'in': 'state:11'}
-        result = self.get('/2015/acs5', params=params)
-        if result.status_code != 200:
-            err = "An error occurred during request: status {0}"
-            raise Exception(err.format(result.status_code))
-        else:
-            data = result
-            print(data.text)
-
-
-CensusApiConn('fakearg').getacs5()
+    def __init__(self, proxies=None):
+        #baseurl not actually used since we need the _urls property to hold many urls. 
+        #Needed to get call to super() to work correctly. TODO refactor so this is optional.
+        baseurl = 'http://api.census.gov/data/'
+        super(CensusApiConn, self).__init__(baseurl, proxies=None)
+        secretsFileName =  os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                               os.pardir,"secrets.json"))    
+        self.api_keys = json.loads(open(secretsFileName).read())
+        self.census_api_key = self.api_keys['census']['api_key']
+
+        self._available_unique_data_ids = [
+                        "acs5_2009",
+                        "acs5_2010",
+                        "acs5_2011",
+                        "acs5_2012",
+                        "acs5_2013",
+                        "acs5_2014",
+                        "acs5_2015",
+                        "acs5_2009_moe",
+                        "acs5_2010_moe",
+                        "acs5_2011_moe",
+                        "acs5_2012_moe",
+                        "acs5_2013_moe",
+                        "acs5_2014_moe",
+                        "acs5_2015_moe"
+                        ]
+
+        self._urls =	{
+                        "acs5_2009"     :"2009/acs5",
+                        "acs5_2010"     :"2010/acs5",
+                        "acs5_2011"     :"2011/acs5",
+                        "acs5_2012"     :"2012/acs5",
+                        "acs5_2013"     :"2013/acs5",
+                        "acs5_2014"     :"2014/acs5",
+                        "acs5_2015"     :"2015/acs5",
+                        "acs5_2009_moe" :"2009/acs5",
+                        "acs5_2010_moe" :"2010/acs5",
+                        "acs5_2011_moe" :"2011/acs5",
+                        "acs5_2012_moe" :"2012/acs5",
+                        "acs5_2013_moe" :"2013/acs5",
+                        "acs5_2014_moe" :"2014/acs5",
+                        "acs5_2015_moe" :"2015/acs5"
+                    	}
+        self._fields = {
+                        "acs5_2009"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E",
```

Comment:
I'll approve this as-is, but as an enhancement it would be great to create the _fields property with a for loop so that we get the same columns for all years as the default behavior, without having to copy paste the list every time. Maybe we can consider this for v2.